### ISSUE TITLE
[Question]: LLMRerank

### ISSUE THREAD

**Issue by AlbertoMQ** [run-llama/llama_index#16595]
### Question Validation

- [X] I have searched both the documentation and discord for an answer.

### Question

        reranker = LLMRerank(
            choice_batch_size=5,
            top_n=reranker_top_n,
            # parse_choice_select_answer_fn=custom_parse_choice_select_answer_fn
        )
        retrieved_nodes = reranker.postprocess_nodes(
            retrieved_nodes, query_bundle
        )

Cell In[9], [line 51](vscode-notebook-cell:?execution_count=9&line=51)
     [42](vscode-notebook-cell:?execution_count=9&line=42)     # reranker = RankLLMRerank(
     [43](vscode-notebook-cell:?execution_count=9&line=43)     #     top_n=reranker_top_n
     [44](vscode-notebook-cell:?execution_count=9&line=44)     # )
   (...)
     [48](vscode-notebook-cell:?execution_count=9&line=48)     #     top_n=reranker_top_n
     [49](vscode-notebook-cell:?execution_count=9&line=49)     # )
     [50](vscode-notebook-cell:?execution_count=9&line=50)     print("num retrieved_nodes:", len(retrieved_nodes))
---> [51](vscode-notebook-cell:?execution_count=9&line=51)     retrieved_nodes = reranker.postprocess_nodes(
     [52](vscode-notebook-cell:?execution_count=9&line=52)         retrieved_nodes, query_bundle
     [53](vscode-notebook-cell:?execution_count=9&line=53)     )
     [55](vscode-notebook-cell:?execution_count=9&line=55) return retrieve
ValueError: invalid literal for int() with base 10: 'The party date is specified in Document 2 with the relevance score of 10. Document 1 should also be consulted as it clarifies the location of the party'

**Comment by logan-markewich** [comment]
You can see, the LLM decided to write a lot of text instead of actually ranking the nodes in a way that's parseable 

What LLM did you use?

**Comment by AlbertoMQ** [comment]
Yes, I noticed that. I was hoping dosu would give me some tips on ensuring a better formatted output from the llm. I'm using a zephyr model. I'm guessing this works better with the larger GPT models.

**Comment by logan-markewich** [comment]
We've come a long way since zephyr for open-source models. Zephyr is a little outdated at this point. Llama3.2 probably does alright with it, along with other recent releases

**Comment by AlbertoMQ** [comment]
I really appreciate your feedback. Do you have any suggestions for models that are available for commercial use? From what I gathered, the llama models are not.
